### PR TITLE
feat(square-connect-plus): migrate to square node.js sdk

### PR DESCRIPTION
**Pull request by tudormd**

### CODE REVIEW

File: src/client/SquareClient.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                throw squareException;
+            }
+        }
+
+        return retry();
+    }
+
+    private isRetryableException(error: SquareApiException): boolean {
+        const squareError: SquareError | undefined = error.errors?.[0];
+        const isRetryableResponseStatusCode: boolean = error.statusCode === 429 || (error.statusCode >= 500 && error.statusCode !== 501);
```
Comment (ruscon) [inline src/client/SquareClient.ts:212]:
```typescript
const isRetryableResponseStatusCode = [408, 429].includes(error.statusCode) || (error.statusCode >= 500 && error.statusCode <= 599 && error.statusCode !== 501);
```

Comment (tudormd) [reply src/client/SquareClient.ts]: fixed

File: test/e2e/client/square-client.spec.ts [hunk 2]
```diff
@@ -98,7 +98,60 @@ describe('SquareClient (e2e)', (): void => {
...
+    it('should NOT retry if in retry config maxRetries is equal with zero', async (): Promise<any> => {
+        nock(customUrl).post(/.*/).times(1000).reply(400);
+
+        return new SquareClient(accessToken, { ...config, retry: { maxRetries: 0 } })
+            .getCustomersApi()
+            .searchCustomers({})
+            .should.eventually.be.rejectedWith(SquareApiException)
+            .and.have.property('retries', 0);
+    });
+    it('should NOT retry if in retry config retryCondition return false', async (): Promise<any> => {
```
Comment (ruscon) [inline test/e2e/client/square-client.spec.ts:134]: \n

File: src/client/SquareClient.ts (hunk 1 above)
Comment (tudormd) [reply src/client/SquareClient.ts]: fixed

File: test/e2e/client/square-client.spec.ts (hunk 2 above)
Comment (tudormd) [reply test/e2e/client/square-client.spec.ts:134]: Ok, will be fix

**Review by ruscon — APPROVED** [review]

**Pull request merged**